lib/src/dependency.dart

```diff
@@ -184,9 +189,9 @@ class HostedDependency extends Dependency {
   final VersionConstraint version;
 
   @JsonKey(disallowNullValue: true)
```

Comment:
How does the `disallowNullValue` work with the nullable field? Something seems a bit off. Some more of these below as well, probably worth a comment at least if this is accomplishing something.

Reply:
I should just get rid of that. Sometimes you want to allow an empty `key` (no value) instead of explicitly null.

In this case, it's not helpful.

Reply:
`disallowNullValue` stops folks from doing `host: # nothing here`

That's all.

The other cases of `required: true, disallowNullValue: true` need to be fixed in checkedYaml

Reply:
https://github.com/google/json_serializable.dart/pull/818 going to wait until this lands and I publish